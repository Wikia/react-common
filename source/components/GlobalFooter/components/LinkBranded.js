import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import I18nNamespaceContext from '../context/I18nNamespaceContext';

const LinkBranded = ({ model }) => {
    const [t] = useTranslation(useContext(I18nNamespaceContext));

    return (
        <a href={model.href} className="wds-global-footer__link" data-tracking-label={model.tracking_label}>
            <div>
                {t(model.title.key)}
            </div>
        </a>
    );
};

LinkBranded.propTypes = {
    model: PropTypes.shape().isRequired,
};

export default LinkBranded;
