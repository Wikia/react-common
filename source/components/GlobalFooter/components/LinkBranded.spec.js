import React from 'react';
import renderer from 'react-test-renderer';

import LinkBranded from './LinkBranded';

test('LinkBranded renders correctly', () => {
    const component = renderer.create(
        <LinkBranded model={{
            type: 'link-branded',
            brand: 'explore-wikis',
            title: {
                type: 'translatable-text',
                key: 'global-footer-fandom-overview-link-explore-wikis',
            },
            href: 'http://fandom.wikia.com/explore',
        }}
        />,
    );

    expect(component.toJSON()).toMatchSnapshot();
});
