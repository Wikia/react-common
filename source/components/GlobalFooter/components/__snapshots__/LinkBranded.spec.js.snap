// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkBranded renders correctly 1`] = `
<a
  className="wds-global-footer__link"
  href="http://fandom.wikia.com/explore"
>
  <div>
    global-footer-fandom-overview-link-explore-wikis
  </div>
</a>
`;
